### ISSUE TITLE
Some UI suggestions

### ISSUE THREAD

**Issue by offset-torque** [koreader/koreader#12362]
I am updating the guide so I am testing the new features now. 
These are some small problems I noticed:

### **Bookmarks > Show in items** menu
There are three items in this new menu currently:
- Highlighted text
- Highlighted text and note
- Note, or highlighted text

First two items are understandable but they would be clearer like this I think. We have space for an extra word:
- Only highlighted text
- Both highlighted text and note

But third item is the real problem here. Because title doesn't describe what it is doing. I enabled it and it showed **ONLY** the note. I think the logic is **"Show the note if it exists, otherwise show the higlighted text"**. This should be communicated better so I am inviting @Frenzie to have a look.

### **New slim skim widget**
Buttons are too narrow vertically @hius07. Even though I don't have fat fingers I had to carefully aim for the buttons. This is really not necessary. Our normal skim widget has properly sized buttons. If we make these slim skim widget buttons square (height equal to current width) they will have the same size as the normal skim widget buttons, which are much better as a touch target. And we will lose only one line of text which is not significant compared to the user benefit.

---

**NOTE:** I am not requesting these changes for myself, these are general UI improvements for the benefit of other users. I don't take notes and I use **Book map** to move around.

**Comment by hius07** [comment]
> Buttons are too narrow vertically

It's here. What is a good height?
https://github.com/koreader/koreader/blob/6de2a59bd88ec67511e269191addda5e1af2c588/frontend/ui/widget/skimtowidget.lua#L71

**Comment by hius07** [comment]
> I think the logic is **"Show the note if it exists, otherwise show the higlighted text"**.

Exactly.

**Comment by offset-torque** [comment]
> What is a good height?

We are limited here by the width actually. Buttons can't be higher than its width (it would look weird) so we can increase the height until they are square as I said. 

These new buttons look 8.4mm wide on my screen. And their height is 4.2mm, exactly half. If there are no objections to the square shape, I suggest making the height same as width. Extra height will help with the crowded button layout. 

If square shape is problematic, it should be at least the same height as normal skim widget buttons which are 6mm tall on my screen. And this is around the lower limit for a "comfortable" touch target.

**Comment by hius07** [comment]
Excuse me, my reference to the buttons height was wrong. Actually we can change the font size to 18, looks good.
https://github.com/koreader/koreader/blob/6de2a59bd88ec67511e269191addda5e1af2c588/frontend/ui/widget/skimtowidget.lua#L65

**Comment by Frenzie** [comment]
> But third item is the real problem here. Because title doesn't describe what it is doing. I enabled it and it showed ONLY the note. I think the logic is "Show the note if it exists, otherwise show the higlighted text". This should be communicated better so I am inviting @Frenzie to have a look.

Roughly what you said?

`Note if available, otherwise highlighted text` (or `fall back to`)

**Comment by offset-torque** [comment]
> Actually we can change the font size to 18, looks good.

I tried 18, much better. I think this is the maximum height we can go while keeping the aesthetics.

> Note if available, otherwise highlighted text

This sounds very clear.

**Comment by poire-z** [comment]
> Note if available, otherwise highlighted text

I don't feel like "available" is the right word for a user added note.
`Note if any, otherwise highlighted text`
`Note if set, otherwise highlighted text`

**Comment by Frenzie** [comment]
I like `set`.

**Comment by offset-torque** [comment]
> I don't feel like "available" is the right word for a user added note.

Makes sense.

> I like set.

+1

**Comment by offset-torque** [comment]
### Some notes about Collections feature

- Collections main dialog has 4 actions divided in two menus: Hamburger menu (create, sort) and long-press menu (rename, delete). Two different places for just four items is not optimal. Users have to remember which item is in which menu, so keeping everything in a single menu is more user friendly in this case. Since rename and delete actions needs a target, we can't move them to hamburger menu. Remaining option is moving hamburger menu items to long press menu. If we do this I can roughly say this: **"You can perform all Collection actions by pressing long on any collection"**. Users don't need to learn or guess the items (and we already have a **Favorites** collection to press on). You (plural) might consider something like this. Although this is not a big deal, since users will eventually get used to split menu and split is logical.

- This one is trickier. I just wrote this to the guide and noticed the weirdness: "To **remove** a book from a collection, long press on it and choose **Add to collection**". This is very unintuitive. After adding a book to a collection, users will try to remove and they will expect a **Remove from collection** button (I understand the reason it is not like this-multiple membership). Of course they will press the **Add to collection** button eventually but renaming this button is a good idea. I don't have a good suggestion. This button works as **Collection management** but this is too long and not that descriptive. **Add/remove from collection** is descriptive but also long. Collection options, collection actions, some icon(s)? So if anyone has a better idea, we can avoid some confusion here.

**Comment by hius07** [comment]
> * the **Add to collection** button

I started from <kbd>Book in collections</kbd>, it was rejected and discussed through https://github.com/koreader/koreader/pull/11693.
No better ideas than <kbd>Add to collection</kbd> proposed.

**Comment by hius07** [comment]
> * Collections main dialog has 4 actions divided in two menus

I'm inclined to keep this separation, as long as we have it in other lists.
Context menu for a certain element and titlebar menu for the whole list.

**Comment by jonnyl2** [comment]
Increasing the button size on the compact SkimTo widget is good, but on fixed-layout docs it is now even more likely to cover up things like page numbers or chapter names in scanned books (which are often cropped out during reading, but nice to see while skimming the book). To remedy this, could one of the following solutions be considered:

- Show the entire page above/below the compact widget (no overlap)
- Allow toggling the compact widget's position between top and bottom of screen during use (e.g., long press on  ↩︎ button)

(This is not an issue on reflowable docs as the covered up lines are unlikely to contain information particularly important while skimming; and even if they do do, the compact skim-to widget can quickly and conveniently be closed and reopened without full-screen refresh.)

**Comment by hius07** [comment]
It will be movable with a swipe (north/south).

**Comment by hius07** [comment]
If there are no good ideas to rename <kbd>Add to collection</kbd>, this can be closed I believe.

**Comment by jonnyl2** [comment]
>If there are no good ideas to rename Add to collection, this can be closed I believe.

Personally, I would name it <kbd>Collections…</kbd>. The ellipsis implies that a new window will open where a selection can be made. And it is neutral as to adding or removing from collections (as both work from here).

**Comment by Frenzie** [comment]
I'm on board with that. I don't know if an ellipsis is necessary.

**Comment by offset-torque** [comment]
> Personally, I would name it Collections… The ellipsis implies that a new window will open where a selection can be made. 

I like this

Also this topic doesn't need to stay open **hius07** since it is not an actionable FR exactly. I can add other notes here if necessary while testing for the user guide.

**Comment by jonnyl2** [comment]
>I don't know if an ellipsis is necessary.

I guess not strictly necessary, but clearer. Without the ellipsis it could just lead to an information screen. The ellipsis signifies that further options will become available. (As in <kbd>Open with...</kbd>)

**Comment by offset-torque** [comment]
In the new compact skim widget, there is this circular icon which I think is "go back to initial location where widget is opened". Anyone who has used a browser will register this as a **Refresh** button. I suggest replacing that with a more suitable arrow that indicates this action better like these examples:
- ⮌ (nearly universal **Back/Previous** icon on all kinds of electronic devices)
- ⤺(looks like a jump back action which might be suitable here)

**Comment by offset-torque** [comment]
I strongly disagree **jonnyl2**. You are demonstrating the expert bias here. You are a programmer who knows the function of that button so you are reverse rationalizing the icon.

I assure you no normal user thinks like this:
 **"That button looks like a refresh button, I bet it will refresh the initial state of the widget."**

 "Initial state" and "widget" words are not in the daily vocabulary of an average e-reader user. 

It will be more like this:
**"That's a refresh button, I wonder what it refreshes."**

And only refresh concept that is related to e-ink devices is the "screen refresh" which is totally unrelated here. So we are showing them a very familiar icon which performs an action totally unrelated to that familiarity and that is a problem.

**Comment by mergen3107** [comment]
Something like a “Home” icon might work better. If icon has this icon, and smaller version of it appears on the progress line itself (maybe with a vertical line), it will make things easier to understand?

Or, as there is an option for showing current session triangle in the bottom progress bar, this arrow can be used here with the same idea?

**Comment by offset-torque** [comment]
I am not sure this issue deserves that much effort **mergen3107**. Because it is really hard to convey this information with just icons **"This button will take you to the page where you first opened this skim dialog."**.

But at least we can invoke the right mental model with a more suitable icon: 
In this case **"This is a back arrow so it will take me back to some previous page"** is a much better guess than **"This is a refresh button so it will refresh something"**.

This icon ⮌ is so universal at this point that even construction machines use it in their UIs:

![tractor_back](https://github.com/user-attachments/assets/5969cc6b-3753-4bec-bb26-50a993049d17)


---

To clarify, none of the problems I report here are important usability problems. They are not even annoyances. They are just minor details and even in the worst case, they just confuse users for a few seconds. So even if they are not addressed, it is not a big deal (This is why they are lumped here together).

But my view is, if we are able to fix these tiny issues (with a reasonable effort), we should, out of respect for users.

**Comment by offset-torque** [comment]
> That's a back button.

Correct that is a back button. It takes you back to where you were before, and everyone recognize this (even construction machine operators :) That is why I am saying it is a better choice here.

Actually this is easy to test. Anyone suspicious about this can print the current icon in the skim widget and ask random people "Do you know this icon?". I am pretty sure answers will be divided between **Refresh** and **Loading**.

**Comment by Frenzie** [comment]
I can't find this icon you're talking about (and you didn't include a screenshot). It's not clear to me if you're talking about a refresh icon or a reset icon.

**Comment by hius07** [comment]
Starting point is marked with a flag sometimes: ⚐ ⚑ ⛳ 🏁 🚩

**Comment by offset-torque** [comment]
> I can't find this icon you're talking about

...in the new compact skim widget, there is this circular icon...

![compact_skim_widget](https://github.com/user-attachments/assets/87307f2d-0c9a-4f70-8acb-dcbadced2d2a)

**Comment by jonnyl2** [comment]
But a flag can often be set and toggled by the user and could be misconstrued by the user as some 'temporary bookmark' while skimming. 

Perhaps this symbol <kbd>▴</kbd> or <kbd>▲</kbd>, which resembles the initial position marker on the progress bar? (Not sure if there's also a symbol with both triangles stacked, pointing to the middle.)

On a side note, this button isn't even really necessary anymore as the same function can be effected via long-press on the page number now (as on the larger widget). The symmetrical design looks nice though, so I wouldn't suggest removing the button, but perhaps there could be a different function assigned altogether?

**Comment by Frenzie** [comment]
> ...in the new compact skim widget, there is this circular icon...

Alright, thanks. Yes, that's a reset icon.

**Comment by offset-torque** [comment]
> But a flag can often be set and toggled by the user and could be misconstrued by the user as some 'temporary bookmark' while skimming.

I agree.

> On a side note, this button isn't even really necessary anymore as the same function can be effected via long-press on the page number now (as on the larger widget). The symmetrical design looks nice though, so I wouldn't suggest removing the button, but perhaps there could be a different function assigned altogether?

If the function is duplicated this option might be more useful. Maybe **Page Browser**? After jumping to an approximate location users might want to check the surrounding pages.

(In this case discoverability is gone of course, users already have to know this long-press action on the page number. But on the other hand, to use all the skim widget functions, users already have to check the guide. Because other buttons have long-press functions but there is no indicator for this.)

Or maybeee this button can be an info button which explains all these functions of the skim widget. This might be more helpful for users.

**Comment by jonnyl2** [comment]
>Maybe Page Browser? After jumping to an approximate location users might want to check the surrounding pages.

I like that idea!

>Or maybeee this button can be an info button which explains all these functions of the skim widget. 

Please no. The scarce space in this little widget shouldn't be wasted on things useful only once...

**Comment by poire-z** [comment]
> Maybe **Page Browser**? After jumping to an approximate location users might want to check the surrounding pages.

They can check the surrounding pages from the... SkimWidget they are already in :)
People are either using mostly PageBrowser to skim, or the SkimWidget - I don't feel we need to make jumping from one to the other in a main button in that SkimWidget, which is best used for something more useful.

And I think the current action "go back to the initial place" is just the right more useful feature.

**Comment by offset-torque** [comment]
Maybe the rewind icon? Kind of makes sense to me, music/video player software have this so it should be familiar and action of going back is roughly similar (movement on a song/book progress bar):

![rewind_icon](https://github.com/user-attachments/assets/7553d9fd-0663-427e-a50a-8566102a5e30)

**Comment by jonnyl2** [comment]
>Maybe the rewind icon?

This would be okay, weren't it for the similarity with the chapter back navigation button. 

How about this one? Often used in maps to show current position.

<img width="86" alt="image" src="https://github.com/user-attachments/assets/85212211-626d-4819-ad2f-347cd7e8c014">

It's in nerdfonts, `EA4D`.

**Comment by offset-torque** [comment]
> How about this one? Often used in maps to show current or initial position.

Yeah I thought about this too, but I wasn't sure if it would make sense to others too. Now I see it makes. Nice looking, familiar icon. +1

**Comment by mergen3107** [comment]
Checkered finish line? 
I don’t know if it is in the KOReader fonts 
https://www.compart.com/en/unicode/U+1F3C1

**Comment by offset-torque** [comment]
> Checkered finish line?

I think that is the opposite of the function here. This button takes us to where we "started" (the widget). There are other flag icons that can be used as a starting flag as **hius07** wrote above, but they are not as recognizable as the checkered one unfortunately.

**Comment by mergen3107** [comment]
(Ah, sorry, I missed those flags. GH in mobile browser is such a mess, but GH mobile app is even worse)

**Comment by solarkite** [comment]
> > I don't feel like "available" is the right word for a user added note.
> 
> Makes sense.
> 
> > I like set.
> 
> +1

Default to note when present/ else set to highlights

Notes default when present/ otherwise set to highlights

**Comment by solarkite** [comment]
> In the new compact skim widget, there is this circular icon which I think is "go back to initial location where widget is opened". Anyone who has used a browser will register this as a **Refresh** button. I suggest replacing that with a more suitable arrow that indicates this action better like these examples:
> 
>     * ⮌ (nearly universal **Back/Previous** icon on all kinds of electronic devices)
> 
>     * ⤺(looks like a jump back action which might be suitable here)

I've been wanting to suggest combining the page browser and Skim for a while. It reduces menu clutter while increasing functionality. I've also included an icon suggestion for the "return to current reading position" function.
![Page skimmer](https://github.com/user-attachments/assets/2bc1c65c-7ae3-47f8-b44b-4a3fbce3afaa)

**Comment by offset-torque** [comment]
### I am summarizing the offered alternatives here for tidyness, in the order of appearance:
- ⮌ ⤺ ⏪   (me)
- ⚐ ⚑ ⛳ 🏁 🚩   (hius07-mergen3107)

- **jonnyl2**
![map_marker](https://github.com/user-attachments/assets/edff8ab5-014f-4fa7-9ec4-c087541f2bd2)

- **solarkite** 
![back_icon](https://github.com/user-attachments/assets/4e4012ea-6b52-4080-9082-c0f118a69619)

**Comment by jonnyl2** [comment]
Let me add another one (the house/home symbol): 
# ⌂

(we probably have more elaborate home symbols in our pictograph sets as well, but this is the simplest one.)

**Comment by solarkite** [comment]
This symbol 
![image](https://github.com/user-attachments/assets/ef7af4b3-19f7-4085-9be6-012db71d7cc8)
"return to current reading position"
combined with a dot positioned at the point on the graph identifying the “current reading position” might be rather intuitive 
![IMG_1905](https://github.com/user-attachments/assets/23956bad-4c90-4dd5-818c-67329f3955c1)
